### PR TITLE
Builds new tkr bom for `v0.12.0`

### LINKED ISSUE #3996: Create TKr BoMs supporting `unmanaged-cluster` in `v0.12.0` release.
* [x] Add secretgen to core repo
* [x] Generate core package repository for `projects.registry.vmware.com/tce/core:v0.12.0`
    * `make generate-package-repo CHANNEL=core TAG=0.12.0`
* [x] Recursively copy core package repo to `projects.registry.vmware.com/tce/core-12`
    * `imgpkg copy -b projects.registry.vmware.com/tce/core:0.12.0 --to-repo projects.registry.vmware.com/tce/core-12`
* [ ] Ensure user-managed package repo is available at `projects.registry.vmware.com/tce/main:v0.12.0`
* [x] Copy kind image from upstream over
    * `crane cp kindest/node:v1.22.7 projects.registry.vmware.com/tce/kind:v1.22.7`
* [x] Build kapp-controller fork for multi-arch support
* [x] Push kapp-controller container image
* [x] Push kapp-controller package
* [ ] Generate new BoM based on `tkr-bom-v1.22.8+vmware.1-tkg.1-tf-v0.11.4.yaml`
    * [x] Reference kind image
    * [ ] Reference core package repo
    * [ ] Reference user-managed package repo
    * [x] Reference kapp-controller package

### PR DESCRIPTION
**Pull request by jpmcb**
## What this PR does / why we need it

New TKr for 0.12.0. Built off of `projects-stg.registry.vmware.com/tkg/tkr-bom:v1.22.8_vmware.1-tkg.1-tf-v0.11.4`

## Which issue(s) this PR fixes
Fixes: #3996

cc @joshrosso

### CODE REVIEW

**Comment by jpmcb** [comment]
Still waiting on user package repo to be available:
```
❯ crane ls projects.registry.vmware.com/tce/main | rg 0.12.0
❯
```

Also @joshrosso - did `repo-12` get pushed?
```
❯ crane ls projects.registry.vmware.com/tce/repo-12
Error: reading tags for projects.registry.vmware.com/tce/repo-12: GET https://projects.registry.vmware.com/v2/tce/repo-12/tags/list?n=1000: NOT_FOUND: repository tce/repo-12 not found
```

**Comment by joshrosso** [comment]
> Still waiting on user package repo to be available:
> 
> ```
> ❯ crane ls projects.registry.vmware.com/tce/main | rg 0.12.0
> ❯
> ```
> 
> Also @joshrosso - did `repo-12` get pushed?
> 
> ```
> ❯ crane ls projects.registry.vmware.com/tce/repo-12
> Error: reading tags for projects.registry.vmware.com/tce/repo-12: GET https://projects.registry.vmware.com/v2/tce/repo-12/tags/list?n=1000: NOT_FOUND: repository tce/repo-12 not found
> ```

I named it wrong :smile: (`core-12`)!

Renaming now for consistency!

File: hack/release/tkr-bom.yaml [hunk 1]
```diff
@@ -1,428 +1,452 @@
...
-  version: 2021.06.28
+  version: 2022.03.30
   thirdPartyImage: true
   osinfo:
     name: ubuntu
     version: "20.04"
     arch: amd64
 imageConfig:
-  imageRepository: projects.registry.vmware.com/tkg
+  imageRepository: projects-stg.registry.vmware.com/tkg
```
Comment (joshrosso) [inline hack/release/tkr-bom.yaml:449]: This needs to be flipped back to `projects.registry.vmware.com/tkg`.

Comment (jpmcb) [reply hack/release/tkr-bom.yaml:449]: Good catch!!!!

**Review by joshrosso — APPROVED** [review]

**Pull request merged**